config/validation.go

```diff
@@ -0,0 +1,30 @@
+package config
+
+func (kafkaConfig KafkaConfig) validateKafkaValues() []string {
+	errs := []string{}
+
+	if len(kafkaConfig.BindAddr) == 0 {
+		errs = append(errs, "no KAFKA_ADDR given")
+	}
+
+	if len(kafkaConfig.MaxBytes) == 0 {
+		errs = append(errs, "no KAFKA_MAX_BYTES given")
+	}
+
+	if len(kafkaConfig.Version) == 0 {
+		errs = append(errs, "no KAFKA_VERSION given")
+	}
+
+	if kafkaConfig.SecProtocol != "" && kafkaConfig.SecProtocol != KafkaTLSProtocolFlag {
+		errs = append(errs, "KAFKA_SEC_PROTO has invalid value")
+	}
+
+	// isKafkaClientCertSet xor isKafkaClientKeySet
+	isKafkaClientCertSet := len(kafkaConfig.SecClientCert) != 0
+	isKafkaClientKeySet := len(kafkaConfig.SecClientKey) != 0
+	if (isKafkaClientCertSet || isKafkaClientKeySet) && !(isKafkaClientCertSet && isKafkaClientKeySet) {
+		errs = append(errs, "only one of KAFKA_SEC_CLIENT_CERT or KAFKA_SEC_CLIENT_KEY has been set - requires both")
+	}
```

Comment:
I think you can simplify this a bit - if SSL is the protocol then both are required fields. If either is missing then its an error - something like:

```
if len(kafkaConfig.SecClientCert) <= 0 {
    // error
}

if len(kafkaConfig.SecClientKey) <= 0 {
    // error
}
```

Reply:
The `SecClientCert` and `SecClientKey` is optional. It doesn't need to be set if TLS is enabled or if it isn't enabled. 
No matter whether enabled or not, if the `SecClientKey` is provided, then `SecClientCert` has to be provided. Likewise, if `SecClientCert` is provided, then `SecClientKey` needs to be provided

Hence why if `isKafkaClientCertSet xor isKafkaClientKeySet`, then an error should be returned

Reply:
The `SecClientCert` and `SecClientKey` is optional. It doesn't need to be set if TLS is enabled or if it isn't enabled. 
No matter whether enabled or not, if the `SecClientKey` is provided, then `SecClientCert` has to be provided. Likewise, if `SecClientCert` is provided, then `SecClientKey` needs to be provided

Hence why if `isKafkaClientCertSet xor isKafkaClientKeySet`, then an error should be returned

Reply:
OK my bad. I still think this could be simplified a bit.
```
if (isKafkaClientCertSet && !isKafkaClientKeySet  || !isKafkaClientCertSet && isKafkaClientKeySet) {
    // error
}
```

Reply:
Done. I have made it a bit more explicit but followed your idea